src/v/raft/offset_translator.cc

```diff
@@ -394,6 +394,14 @@ ss::future<> offset_translator::sync_with_log(
     co_await std::move(reader).for_each_ref(
       log_consumer{*this}, model::no_timeout);
 
+    if (_highest_known_offset < log_offsets.dirty_offset) {
```

Comment:
This assumes that the log had at least one batch in it, right? (otherwise _highest_known_offset would be in its initial ::min() state).  That's usually true but would be nice not to rely on other components only trying to sync with non-empty segments.

Reply:
Actually, offset_translator goes through a rather convoluted startup sequence where it is first `start`-ed, then `prefix_truncate_reset` is performed with data from the raft snapshot, and then it is `sync_with_log`-ed. So before the sync `_highest_known_offset` is at least `_last_snapshot_index`. But I'll check with empty logs.

Reply:
Yeah, for a new partition highest known offset is -infinity and dirty offset is -infinity as well, and if we have no segments, but a valid snapshot, highest known offset is equal to `_last_snapshot_index` which is also the dirty offset.